### DISCUSSION TITLE
Create a one to many and one to one of same model.

### DISCUSSION THREAD

**Discussion by tylermcrobert** [prisma/prisma#23405] [Q&A]
### Question

Hi, I was wondering how I can create a schema that can do the following (simplified) example: 

```prisma
model Track {
  id             String         @id
  allVersions    TrackVersion[]
  primaryVersion Track         
}

model TrackVersion {
  id          String  @id
  track       Track   
  isPrimaryOf Track?  
}
```

How can I model it so when I create a `TrackVersion` and a `Track` together I can assign the newly created `TrackVersion` as primary for the `Track`? 

Thanks in advance, this has been really stumping me.

**Comment by ludralph** [comment]
Hi @tylermcrobert 👋 

Thank you for raising this question.

You can create a one-to-one relationship between `Track` and `TrackVersion` for the `primaryVersion` field, and a one-to-many relationship for the `allVersions` field. Here's how you can define it:

```prisma
model Track {
  id             String         @id
  primaryVersionId String @unique
  primaryVersion TrackVersion   @relation("PrimaryVersion", fields: [primaryVersionId], references: [id])
  allVersions    TrackVersion[] @relation("AllVersions")
}

model TrackVersion {
  id          String  @id
  trackId     String
  track       Track   @relation("AllVersions", fields: [trackId], references: [id])
  primaryTrack Track? @relation("PrimaryVersion")
}
```

In this schema, the `@relation("PrimaryVersion")` and `@relation("AllVersions")` attributes are used to [disambiguate](https://www.prisma.io/docs/orm/prisma-schema/data-model/relations#disambiguating-relations) the relations. The `PrimaryVersion` relation represents the one-to-one relationship between `Track` and `TrackVersion` for the `primaryVersion` field, and the `AllVersions` relation represents the one-to-many relationship for the `allVersions` field.

When creating a new `Track` with a `TrackVersion`, you can assign the newly created TrackVersion as primary for the `Track` by setting the `primaryVersionId` field in the Track model to the `id` of the `TrackVersion`. For example:

```js
async function createTrackWithPrimaryVersion(trackData, trackVersionData) {
  const trackVersion = await prisma.trackVersion.create({
    data: trackVersionData,
  });

  const track = await prisma.track.create({
    data: {
      ...trackData,
      primaryVersionId: trackVersion.id,
      allVersions: {
        create: trackVersionData,
      },
    },
  });

  return track;
}
```
In this function, we first create a new `TrackVersion` record using the `prisma.trackVersion.create()` method. Then, we create a new `Track` record using the `prisma.track.create()` method, and set the `primaryVersionId` field to the `id` of the newly created `TrackVersion`. We also add the new `TrackVersion` to the `allVersions` field.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
